esi_leap/tests/db/sqlalchemy/test_api.py

```diff
@@ -259,4 +266,122 @@ def test_offer_get_by_name(self):
 
     def test_offer_get_by_name_not_found(self):
 
-        assert api.offer_get_by_uuid('some_uuid') is None
+        self.assertEqual(api.offer_get_by_name('some_name'), [])
+
+    def test_offer_destroy(self):
+
+        offer = api.offer_create(test_offer_2)
+        api.offer_destroy(offer.uuid)
+        self.assertEqual(api.offer_get_by_uuid('offer_2'), None)
+
+    def test_offer_destroy_not_found(self):
+
+        self.assertEqual(api.offer_get_by_uuid('offer_4'), None)
+
+    def test_offer_update(self):
+
+        offer = api.offer_create(test_offer_3)
+        o = api.offer_create(test_offer_2)
+        values = {'start_time': o.start_time,
+                  'end_time': o.end_time}
+        api.offer_update(offer.uuid, values)
+        res = api.offer_get_by_uuid(offer.uuid)
+        self.assertEqual(res.start_time, offer.start_time)
+        self.assertEqual(res.end_time, offer.end_time)
+
+    def test_offer_get_all(self):
+
+        o = api.offer_create(test_offer_2)
+        offer = api.offer_create(test_offer_3)
+        res = api.offer_get_all({}).all()
+        self.assertEqual((o.to_dict(), offer.to_dict()),
+                         (res[0].to_dict(), res[1].to_dict()))
+
+    def test_contract_get_by_uuid(self):
+
+        o = api.offer_create(test_offer_2)
+        test_contract_4['offer_uuid'] = o.uuid
+        contract = api.contract_create(test_contract_4)
+        res = api.contract_get_by_uuid(contract.uuid)
+        self.assertEqual(contract.uuid, res.uuid)
+
+    def test_contract_get_by_uuid_not_found(self):
+
+        assert api.contract_get_by_uuid('some_uuid') is None
+
+    def test_contract_get_by_name(self):
+
+        o1 = api.offer_create(test_offer_1)
+        o2 = api.offer_create(test_offer_2)
+        o3 = api.offer_create(test_offer_3)
+        o4 = api.offer_create(test_offer_4)
+        test_contract_1['offer_uuid'] = o1.uuid
+        test_contract_2['offer_uuid'] = o2.uuid
+        test_contract_3['offer_uuid'] = o3.uuid
+        test_contract_4['offer_uuid'] = o4.uuid
+        c1 = api.contract_create(test_contract_1)
+        c2 = api.contract_create(test_contract_2)
+        c3 = api.contract_create(test_contract_3)
+        api.contract_create(test_contract_4)
+        res = api.contract_get_by_name('c1')
+        assert len(res) == 3
```

Comment:
I was confused why you created so many contracts for this, but you're right, this function can return multiple results. Good thinking!